sdk/core/azure-core-amqp/src/test/java/com/azure/core/amqp/implementation/ConnectionOptionsTest.java

```diff
@@ -77,4 +79,48 @@ public void propertiesSet() {
         assertEquals(retryOptions, actual.getRetry());
         assertEquals(verifyMode, actual.getSslVerifyMode());
     }
+
+    /**
+     * Verifies that the correct port and properties are set.
+     */
+    @Test
+    public void propertiesAndPortSet() {
+        // Arrange
+        final String productName = "test-product";
+        final String clientVersion = "1.5.10";
+        final String scope = "test-scope";
+
+        final String fullyQualifiedNamespace = "host-name.com";
+        final SslDomain.VerifyMode verifyMode = SslDomain.VerifyMode.VERIFY_PEER;
+        final AmqpRetryOptions retryOptions = new AmqpRetryOptions();
+        final ClientOptions clientOptions = new ClientOptions();
+        final String actualHostname = "actual-host-name.com";
+        final int port = 1000;
+
+        // Act
+        final ConnectionOptions actual
+            = new ConnectionOptions(fullyQualifiedNamespace, tokenCredential, CbsAuthorizationType.JSON_WEB_TOKEN,
+                scope, AmqpTransportType.AMQP, retryOptions, ProxyOptions.SYSTEM_DEFAULTS, scheduler, clientOptions,
+                verifyMode, productName, clientVersion, actualHostname, port);
+
+        // Assert
+        assertEquals(fullyQualifiedNamespace, actual.getFullyQualifiedNamespace());
+        assertEquals(ConnectionHandler.AMQPS_PORT, actual.getPort());
+        assertEquals(productName, actual.getProduct());
+        assertEquals(clientVersion, actual.getClientVersion());
+
+        assertSame(clientOptions, actual.getClientOptions());
+
+        assertEquals(AmqpTransportType.AMQP, actual.getTransportType());
+        assertEquals(scheduler, actual.getScheduler());
+
+        assertEquals(tokenCredential, actual.getTokenCredential());
+        assertEquals(CbsAuthorizationType.JSON_WEB_TOKEN, actual.getAuthorizationType());
+        assertEquals(scope, actual.getAuthorizationScope());
+        assertEquals(retryOptions, actual.getRetry());
+        assertEquals(verifyMode, actual.getSslVerifyMode());
+
+        assertEquals(actualHostname, actual.getHostname());
+        assertEquals(port, actual.getPort());
```

Comment:
curious; how assert for `getPort()` passes for different port numbers (line_108, line_124)

Reply:
Lol. Good catch. ;) Also a test failure.